src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/AbstractOngoingTasksHandlerProcessorForGetPullReplicationHubTasksInfo.cs

```diff
@@ -0,0 +1,93 @@
+using Raven.Client.Exceptions;
+using Raven.Client.Exceptions.Database;
+using Raven.Client.Http;
+using Raven.Client.ServerWide;
+using Raven.Client.Util;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Handlers.Processors.OngoingTasks
+{
+    internal abstract class AbstractOngoingTasksHandlerProcessorForGetPullReplicationHubTasksInfo<TRequestHandler, TOperationContext> : AbstractDatabaseHandlerProcessor<TRequestHandler, TOperationContext>
+        where TOperationContext : JsonOperationContext
+        where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    {
+        protected AbstractOngoingTasksHandlerProcessorForGetPullReplicationHubTasksInfo([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected abstract IEnumerable<OngoingTaskPullReplicationAsHub> GetOngoingTasks(TransactionOperationContext context, DatabaseRecord databaseRecord, ClusterTopology clusterTopology, long key);
+
+        protected void AssertCanExecute()
+        {
+            if (ResourceNameValidator.IsValidResourceName(RequestHandler.DatabaseName, ServerStore.Configuration.Core.DataDirectory.FullPath, out string errorMessage) == false)
+            {
+                bool sharded = ShardHelper.IsShardedName(RequestHandler.DatabaseName);
+                if (sharded == false)
+                    throw new BadRequestException(errorMessage);
+            }
+        }
+
+        public override async ValueTask ExecuteAsync()
+        {
+            AssertCanExecute();
+
+            var key = RequestHandler.GetLongQueryString("key");
+
+            using (RequestHandler.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                ClusterTopology clusterTopology;
+                PullReplicationDefinition def;
+                using (context.OpenReadTransaction())
+                {
+                    clusterTopology = RequestHandler.ServerStore.GetClusterTopology(context);
+                    using (var rawRecord = RequestHandler.ServerStore.Cluster.ReadRawDatabaseRecord(context, RequestHandler.DatabaseName))
+                    {
+                        if (rawRecord == null)
+                            throw new DatabaseDoesNotExistException(RequestHandler.DatabaseName);
+
+                        def = rawRecord.GetHubPullReplicationById(key);
+                    }
+
+                    if (def == null)
+                    {
+                        HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                        return;
+                    }
+
+
+                    var databaseRecord = RequestHandler.ServerStore.Cluster.ReadDatabase(context, RequestHandler.DatabaseName);
+
+                    var currentHandlers = GetOngoingTasks(context, databaseRecord, clusterTopology, key);
```

Comment:
you have 'rawRecord' few lines up? Use it or use rawRecord.MateralizedRecord ?

Reply:
done